pkg/controller/core/clusterqueue_controller_test.go

```diff
@@ -203,3 +208,273 @@ func TestUpdateCqStatusIfChanged(t *testing.T) {
 		})
 	}
 }
+
+type cqMetrics struct {
+	NominalDPs   []testingmetrics.GaugeDataPoint
+	BorrowingDPs []testingmetrics.GaugeDataPoint
+	UsageDPs     []testingmetrics.GaugeDataPoint
+}
+
+func allMetricsForQueue(name string) cqMetrics {
+	return cqMetrics{
+		NominalDPs:   testingmetrics.CollectFilteredGaugeVec(metrics.ClusterQueueResourceNominalQuota, map[string]string{"cluster_queue": name}),
+		BorrowingDPs: testingmetrics.CollectFilteredGaugeVec(metrics.ClusterQueueResourceBorrowingLimit, map[string]string{"cluster_queue": name}),
+		UsageDPs:     testingmetrics.CollectFilteredGaugeVec(metrics.ClusterQueueResourceUsage, map[string]string{"cluster_queue": name}),
+	}
+}
+
+func testDataPoint(cohort, name, flavor, res string, v float64) testingmetrics.GaugeDataPoint {
+	return testingmetrics.GaugeDataPoint{
+		Labels: map[string]string{
+			"cohort":        cohort,
+			"cluster_queue": name,
+			"flavor":        flavor,
+			"resource":      res,
+		},
+		Value: v,
+	}
+}
+
+func TestRecordResourceMetrics(t *testing.T) {
+	baseQueue := &kueue.ClusterQueue{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "name",
+		},
+		Spec: kueue.ClusterQueueSpec{
+			Cohort: "cohort",
+			ResourceGroups: []kueue.ResourceGroup{
+				{
+					CoveredResources: []corev1.ResourceName{corev1.ResourceCPU},
+					Flavors: []kueue.FlavorQuotas{
+						{
+							Name: "flavor",
+							Resources: []kueue.ResourceQuota{
+								{
+									Name:           corev1.ResourceCPU,
+									NominalQuota:   resource.MustParse("1"),
+									BorrowingLimit: pointer.Quantity(resource.MustParse("2")),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Status: kueue.ClusterQueueStatus{
+			FlavorsUsage: []kueue.FlavorUsage{
+				{
+					Name: "flavor",
+					Resources: []kueue.ResourceUsage{
+						{
+							Name:     corev1.ResourceCPU,
+							Total:    resource.MustParse("2"),
+							Borrowed: resource.MustParse("1"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testCases := map[string]struct {
+		queue          *kueue.ClusterQueue
+		metrics        cqMetrics
```

Comment:
Always prefix with `want` things that we check

```suggestion
		wantMetrics        cqMetrics
```

Reply:
done